requirements/specifications/intents/user-interest.md

```diff
@@ -0,0 +1,290 @@
+requirements anywhere?
+
+## 4. User Interest from a platform UX
+
+Firebolt platforms may provide a platform UX, e.g. voice or and RCU, to
+express user interest in content from an active App. To facilitate this
+Apps will need to be told about the user\'s expressed interest in their
+content.
+
+First, the Aggregated Experience (or some app with the correct
+capability) detects that the user is interested in something. In this
+picture the interest is triggered by an RCU button, but how this occurs
+is outside the scope of this document. When this happens, the Aggregated
+Experience app calls `Content.requestUserInterest()`, which will trigger the
+foreground app's UserInterest provider and call it's `userInterest` method
+by invoking the RPC method `Discovery.onRequestUserInterest`.
+
+![](../../../requirements/images/specifications/intents/user-interest/media/image3.png)
+
+Next, the foreground app receives and responds to the request with an 
+EntityInfo, which is returned as the result to the pending
+`Content.requestUserInterest` method:
+
+![](../../../requirements/images/specifications/intents/user-interest/media/image4.png)
+
+A user's intention to express interest in something is handled by the
+`InterestIntent`. This intent happens *before* the platform knows which
+content is displayed and/or selected by the foreground app. The
+`InterestIntent` may originate from a cloud system, e.g. Voice, or from the RCU, or elsewhere. Origination of this intent is out
+of scope for this document. This intent includes the type of interest,
+e.g. interest or disinterest.
+
+To fire an `InterestIntent` to the current App, another app, typically
+the Aggregated Experience, will call the `Content.requestUserInterest()` method
+of the Discovery SDK.
+
+When `Content.requestUserInterest` is called, if the App in the foreground 
+Lifecycle state has registered a provider for the `xrn:firebolt:capability:discovery:interest` capability, then it **MUST** have that provider invoked with the values from the
+`Content.requestUserInterest` call.
+
+Once an App's callback is invoked, that app will have `interestTimeout`
+milliseconds to return a value or throw an error. Values returned after
+that time **MUST** be ignored. The timeout value is stored in the
+device's configuration manifest.
+
+To be notified when a user expresses interest in the currently displayed
+content, an App **MUST** provide the
+`xrn:firebolt:capability:discovery:interest` capability by enabling the
+`Discovery.onRequestUserInterest` notification.
+
+If there is a valid entity to return, then the method registered by the
+App **MUST** return the currently displayed entity meta-data.
+
+If there is no valid entity to return, then the method **MUST** throw an
+exception.
+
+If the foreground App returns a valid `EntityInfo` before the timeout,
+then, the value from the **MUST** be used.
+
+Once a single `EntityInfo` is selected the platform **MUST** dispatch a
```

Comment:
Think this paragraph is wrong, this section is the "pull" method so it should just be responding to the requestInterest method, not dispatch a new event.

Reply:
Good catch. I cleaned this up and moved it to the push section.